tests/run-make/issue-125484-used-dependencies/rmake.rs

```diff
@@ -0,0 +1,18 @@
+// Non-regression test for issues #125474, #125484, #125646, with the repro taken from #125484. Some
+// queries use "used dependencies" while others use "speculatively loaded dependencies", and an
+// indexing ICE appeared in some cases when these were unexpectedly used in the same context.
+
+// FIXME: this should probably be a UI test instead of a run-make test, but I *cannot* find a way to
+// make compiletest annotations reproduce the ICE with the minimizations from issues #125474 and
+// #125484.
```

Comment:
(UI tests compile the main test file and auxiliary crates with extra flags, especially having to do with library search paths and how the auxiliary crates are linked in and whatnot, so I think it is fine that this stays as a run-make test.)

Reply:
Yeah and it's not clear that these extra flags are what's causing the repro to disappear, so some judicious compile-flags could still work but I don't have the time to look into that at the moment.

Reply:
@jieyouxu actually rmake tests also add an -L flag to rustc's invocations. 

In any case, it's similarly not straightforward to get the #125474 MCVE to ICE _in an rmake test_, I'm not sure what's going on. I'd like to avoid using a makefile.